pulsar-client-cpp/CMakeLists.txt

```diff
@@ -23,13 +23,24 @@ project (pulsar-cpp)
 option(LINK_STATIC "Link against static libraries" OFF)
 MESSAGE(STATUS "LINK_STATIC:  " ${LINK_STATIC})
 
+option(DEBUG_MODE "Enable debug mode" OFF)
```

Comment:
Can we try using 

cmake CMAKE_BUILD_TYPE=Debug .

It has worked for me in the past